template/app/src/main/java/co/nimblehq/template/ui/screens/xml/HomeViewModel.kt

```diff
@@ -31,6 +39,27 @@ class HomeViewModel @Inject constructor(
                     _uiModels.emit(uiModels)
                 }
             hideLoading()
+
+            requestPermissions()
+        }
+    }
+
+    private suspend fun requestPermissions() {
+        _requestPermissions.emit(
+            arrayOf(
+                Manifest.permission.ACCESS_FINE_LOCATION,
+                Manifest.permission.CAMERA
+            )
+        )
+    }
+
+    fun onPermissionResult(permissionResult: PermissionResult) {
+        when (permissionResult) {
+            is Granted -> Timber.d("${permissionResult.grantedPermissions} granted")
+            is Denied.JustDenied -> Timber.d("${permissionResult.deniedPermissions} denied")
+            is Denied.NeedsRationale -> Timber.d("${permissionResult.deniedPermissions} needs rationale")
+            is Denied.DeniedPermanently -> Timber.d("${permissionResult.deniedPermissions} denied for good")
+            is Cancelled -> Timber.d("request cancelled")
```

Comment:
Same here, I'm wondering if we should only keep the dependency in `template` and move the rest into the `samples` 

What do you think? 🙏